syntax/scan.go

```diff
@@ -462,11 +473,18 @@ start:
 
 	// comment
 	if c == '#' {
+		sc.startToken(val)
 		// Consume up to (but not including) newline.
 		for c != 0 && c != '\n' {
 			sc.readRune()
 			c = sc.peekRune()
 		}
+		sc.endToken(val)
```

Comment:
Skip this unless comments are requested.  It allocates a string for the comment.